services/case_filters/case_filter_base.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from abc import abstractmethod
+from typing import List
+
+from models.blaise.blaise_case_information_model import BlaiseCaseInformationModel
+from models.totalmobile.totalmobile_world_model import TotalmobileWorldModel
+
+
+class CaseFilterBase:
+    @property
+    @abstractmethod
+    def wave_number(self) -> int:
+        pass
+
+    @abstractmethod
+    def case_is_eligible(self, case: BlaiseCaseInformationModel) -> bool:
+        pass
+
+    @staticmethod
+    def case_is_part_of_wave(
+        wave_number: int, case: BlaiseCaseInformationModel
```

Comment:
If this method wasn't static you could just use `self.wave_number` rather than pass it in. This is the only valid reason I can see for the `wave_number` to exist at all, in its current state it doesn't seem to be used anywhere which required it to be an abstract property?

Reply:
It is used in the first method call case_is_part_of_wave which is integral as to how we agree the filtering works - if the case is not part of the wave it moves on to the next filter

Reply:
It's only used in the child classes though, so it's technically just an implementation detail of those classes as it's not accessed in a polymorphic way anywhere (unless I've not seen it somewhere?)

Reply:
I'm referring to the `wave_number` property above, not the `case_is_part_of_wave` method. I `case_is_part_of_wave` switched to using `self.wave_number` then both would be valid to exist.